src/metabase/models/notification.clj

```diff
@@ -391,3 +428,29 @@
               handler-id (t2/insert-returning-pk! :model/NotificationHandler handler)]
           (t2/insert! :model/NotificationRecipient (map #(assoc % :notification_handler_id handler-id) recipients))))
       instance)))
+
+(def ^:private notification-update-spec
+  {:model        :model/Notification
+   ;; a function that takes a row and returns a map of the columns to compare
+   :compare-row  #(select-keys % [:active])
+   :nested-specs {:payload       {:model        :model/NotificationCard
+                                  :compare-row  #(select-keys % [:send_condition :send_once])}
```

Comment:
I'd much rather the spec itself was pure-data.

```suggestion
                                  :compare-values [:send_condition :send_once]}
```

Reply:
+1 here, at least until there is an obvious need for custom processing

Also maybe not `:compare-values`, but something like `:row-id` or `:unique-id`?

Reply:
how about :compare-cols?